src/main/java/edu/cornell/kfs/pmw/batch/dataaccess/impl/PaymentWorksVendorDaoJdbc.java

```diff
@@ -2,87 +2,122 @@
-        sql.append(" proc_ts = '" + PROCESSING_TIMESTAMP_SQL_FORMATTER.format(processingTimeStamp) + "'");
+
+        sqlFactory.appendSql(" kfs_vnd_proc_stat = ?,", kfsVendorProcessingStatus);
+        sqlFactory.appendSql(" proc_ts = ?", PROCESSING_TIMESTAMP_SQL_FORMATTER.format(processingTimeStamp));
+
         if (StringUtils.isNotBlank(kfsAchProcessingStatus)) {
-            sql.append(", kfs_ach_proc_stat = '" + kfsAchProcessingStatus + "'");
+            sqlFactory.appendSql(", kfs_ach_proc_stat = ?", kfsAchProcessingStatus);
         }
         if (StringUtils.isNotBlank(supplierUploadStatus)) {
-            sql.append(", supp_upld_stat = '" + supplierUploadStatus + "'");
+            sqlFactory.appendSql(", supp_upld_stat = ?", supplierUploadStatus);
         }
         if (StringUtils.isNotBlank(kfsVendorDocumentNumber)) {
-            sql.append(", pven_fdoc_nbr = '" + kfsVendorDocumentNumber + "'");
+            sqlFactory.appendSql(", pven_fdoc_nbr = ?", kfsVendorDocumentNumber);
         }
         if (ObjectUtils.isNotNull(vendorHeaderGeneratedIdentifier)) {
-            sql.append(", vndr_hdr_gnrtd_id = '" + vendorHeaderGeneratedIdentifier.intValue() + "'");
+            sqlFactory.appendSql(", vndr_hdr_gnrtd_id = ?", vendorHeaderGeneratedIdentifier);
         }
         if (ObjectUtils.isNotNull(vendorDetailAssignedIdentifier)) {
-            sql.append(", vndr_dtl_asnd_id = '" + vendorDetailAssignedIdentifier.intValue() + "'");
+            sqlFactory.appendSql(", vndr_dtl_asnd_id = ?", vendorDetailAssignedIdentifier);
         }
         if (ObjectUtils.isNotNull(kfsAchDocumentNumber)) {
-            sql.append(", paat_fdoc_nbr = '" + kfsAchDocumentNumber + "'");
+            sqlFactory.appendSql(", paat_fdoc_nbr = ?", kfsAchDocumentNumber);
         }
-        sql.append(" where id = '" + id + "'");
-        
-        return sql.toString();
+        sqlFactory.appendSql(" where id = ?", id);
+
+        JdbcTemplate jdbcTemplate = getJdbcTemplate();
+        return jdbcTemplate.update(sqlFactory.getSql(), sqlFactory.getParameters().toArray());
     }
 
+    private class ParameterizedSqlFactory {
+        private StringBuilder sqlStringBuilder;
+        private List<Object> parameters;
+
+        public ParameterizedSqlFactory(String initialSql) {
+            sqlStringBuilder = new StringBuilder(initialSql);
+            parameters = new ArrayList<>();
+        }
+
+        public void appendSql(String sql) {
+            sqlStringBuilder.append(sql);
+        }
+
+        public void appendSql(String sql, Object parameter) {
+            appendSql(sql);
+            parameters.add(parameter);
+        }
+
+        public String toString() {
+            String effectiveSql = sqlStringBuilder.toString();
+            for (Object parameter : parameters) {
+                effectiveSql = effectiveSql.replaceFirst("\\?", parameter instanceof Integer ? ("%d") : "'%s'");
```

Comment:
This could still be reworked to be more efficient. However, given that there does not appear to be any code that is actively using this method, I am okay with leaving it as-is for now, and it can be reworked later if necessary.